src/_pytest/logging.py

```diff
@@ -468,6 +469,16 @@ def _setup_cli_logging(self):
             log_cli_handler, formatter=log_cli_formatter, level=log_cli_level
         )
 
+    def set_log_path(self, fname):
+        if isinstance(fname, pathlib.PurePath):
+            fname = str(fname)
+
+        if not os.path.exists(os.path.dirname(fname)):
```

Comment:
Following my advice above, it is just a matter of changing this to `fname.parent.mkdir(exist_ok=True, parents=True)`

Reply:
Thanks, fixed